src/main/java/org/wildfly/security/authz/MappedRoleMapper.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.authz;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A simple mapping role mapper.
+ *
+ * Maps each role to a set of new roles using a String to Set<String> map.
+ *
+ * @author <a href="mailto:mmazanek@redhat.com">Martin Mazanek</a>
+ */
+public class MappedRoleMapper implements RoleMapper {
+
+    private final Map<String, Set<String>> reverseRoleMap;
+
+    /**
+     * Construct a new instance.
+     * Called from WildFly core when using this as a custom component. You should not use this constructor and use {@link MappedRoleMapper.Builder} instead.
+     *
+     * @see MappedRoleMapper.Builder
+     */
+    public MappedRoleMapper() {
+        this.reverseRoleMap = new LinkedHashMap<>();
```

Comment:
maybe, would not be keeping the map null better, to ensure uninitialized mapper will not be used accidentally?

Reply:
(or at least, if you want to keep the map final, to empty the map on initialization, to prevent unexpected behavior if `initialize` would be called multiple times...)